datafusion/core/src/physical_planner.rs

```diff
@@ -777,6 +777,16 @@ impl DefaultPhysicalPlanner {
                     .map(|s| (s.to_string(), arrow_schema::DataType::Null))
                     .collect::<Vec<_>>();
 
+                let keep_partition_by_columns = source_option_tuples
+                    .get("hive.keep_partition_by_columns")
+                    .map(|v| v.trim() == "true")
+                    .unwrap_or(false)
+                    || session_state
+                        .config()
```

Comment:
I believe using "OR" here is a bit risky. Explicitly setting `hive.keep_partition_by_columns` to `false` and not specifying anything in `options` would behave the same way. For other format options in similar scenarios, we assume that if an option is explicitly set (`true` or `false`), we take that value. Otherwise, if nothing is specified, we consult the `session_config` value. I suggest applying a similar behavior for this option as well.

Reply:
Agreed, if the user is explicit setting false, we should respect that. I'll change the code to do:

```
        let keep_partition_by_columns = source_option_tuples
                    .get("hive.keep_partition_by_columns")
                    .map(|v| v.trim() == "true")
                    .unwrap_or(session_state
                        .config()
                        .options()
                        .execution
                        .keep_partition_by_columns);
```